TimberAPI/AssetLoaderSystem/ResourceAssetPatch/ResourceAssetLoaderTimberApi.cs

```diff
@@ -0,0 +1,80 @@
+using Bindito.Core;
+using HarmonyLib;
+using Timberborn.AssetSystem;
+using TimberbornAPI.AssetLoaderSystem.AssetSystem;
+using UnityEngine;
+using Object = UnityEngine.Object;
+
+namespace TimberbornAPI.AssetLoaderSystem.ResourceAssetPatch
+{
+    public class ResourceAssetLoaderTimberApi : IResourceAssetLoader
+    {
+        private readonly IAssetLoader _assetLoader;
+
+        private Shader _shader;
+
+        private Shader Shader
+        {
+            get
+            {
+                if (_shader == null)
+                    _shader = Resources.Load<GameObject>("Buildings/Paths/Platform/Platform.Full.Folktails").GetComponent<MeshRenderer>().materials[0].shader;
+                return _shader;
+            }
+        }
+
+        public ResourceAssetLoaderTimberApi(IAssetLoader assetLoader)
+        {
+            _assetLoader = assetLoader;
+        }
+
+        public T Load<T>(string path) where T : Object
+        {
+            T item = Resources.Load<T>(path);
+
+            if (item != null)
+                return item;
+
+            item = _assetLoader.Load<T>(path);
+            if (item is GameObject gameObject)
+                FixMaterialShader(gameObject, Shader);
+
+            return item;
+        }
+
+        public T[] LoadAll<T>(string path) where T : Object
+        {
+            return Resources.LoadAll<T>(path);
+        }
+
+        private static void FixMaterialShader(GameObject obj, Shader shader)
+        {
+            var meshRenderer = obj.GetComponent<MeshRenderer>();
+            if (meshRenderer)
+            {
+                foreach (Material mat in meshRenderer.materials)
+                {
+                    mat.shader = shader;
+                }
+            }
+
+            foreach (Transform child in obj.transform)
+            {
+                if (child.gameObject)
+                {
+                    FixMaterialShader(child.gameObject, shader);
+                }
+            }
+        }
+    }
+    
+    [HarmonyPatch(typeof(AssetSystemConfigurator), "Configure", typeof(IContainerDefinition))]
```

Comment:
This could be moved to it's own file called "TimberApiResourceAssetLoaderPatch" or something, to be in line with how there are already some files just for the patches.